File: test/easy/FindContentChildrenTest.php

<?php

use Src\Easy\FindContentChildren;

describe('Feature Find Content Children', function () {
    $instance = new FindContentChildren;
    describe('Scenario:', function () use ($instance) {
        test('Testing', function ($children, $cookies, $expected) use ($instance) {
            expect($instance->solve($children, $cookies))->toBe($expected);
        })->with([
            'some children unsatisfied' => [[1, 2, 3], [1, 1], 1],
            'all children satisfied' => [[1, 2], [1, 2, 3], 2],
            'unsorted children, sorted cookies' => [[10, 9, 8, 7], [5, 6, 7, 8], 2],
            'no children' => [[], [1, 2], 0],
            'no cookies' => [[1, 2], [], 0],
        ]);
    });
});

File: test/easy/find_content_children_test.dart
import 'package:test/test.dart';
import 'package:dark_coding_challenges/easy/find_content_children.dart';

class FindContentChildrenTestCase {
  List<int> children;
  List<int> cookies;
  int expected;
  FindContentChildrenTestCase(this.children, this.cookies, this.expected);
}

void main() {
  group("Feature: Find Content Children Challenge", () {
    final testCases = [
      FindContentChildrenTestCase([1, 2, 3], [1, 1], 1),
      FindContentChildrenTestCase([1, 2], [1, 2, 3], 2),
      FindContentChildrenTestCase([10, 9, 8, 7], [5, 6, 7, 8], 2),
      FindContentChildrenTestCase([], [1, 2], 0),
      FindContentChildrenTestCase([1, 2], [], 0),
    ];

    for (final step in testCases) {
      // Safely cast values to List<int> for nums and expected
      final children = step.children;
      final cookies = step.cookies;
      final expected = step.expected;

      test("Testing is the children $children\x1b[0m can be satisfied by cookies $cookies\x1b[0m", () {
        int result = findContentChildren(children, cookies);
        expect(result, equals(expected));
      });
    }
  });
}
